src/Dfe.PlanTech.Infrastructure.Data/CmsDbContext.cs

```diff
@@ -252,14 +252,42 @@ private static string GetCacheKey(IQueryable query)
     public async Task<List<T>> ToListAsync<T>(IQueryable<T> queryable, CancellationToken cancellationToken = default)
     {
         var key = GetCacheKey(queryable);
-        return await _queryCacher.GetOrCreateAsyncWithCache(key, queryable,
+        var result = await _queryCacher.GetOrCreateAsyncWithCache(key, queryable,
             (q, ctoken) => q.ToListAsync(ctoken), cancellationToken);
+
+        if (result != null)
+        {
+            //Attach the result to the DbContext change tracker.
+            //If the result was retrieved from the cache, or it is going to be used by another entity that _was_ retrieved from the cache,
+            //Then we need to ensure that EF Core is aware of it, otherwise it will not fix-up the navigations as it would normally
+            foreach (var item in result)
+            {
+                AttachEntity(item);
+            }
+        }
+
+        return result ?? new List<T>();
     }
 
     public async Task<T?> FirstOrDefaultAsync<T>(IQueryable<T> queryable, CancellationToken cancellationToken = default)
     {
         var key = GetCacheKey(queryable);
-        return await _queryCacher.GetOrCreateAsyncWithCache(key, queryable,
+        var result = await _queryCacher.GetOrCreateAsyncWithCache(key, queryable,
             (q, ctoken) => q.FirstOrDefaultAsync(ctoken), cancellationToken);
+
+        //Attach the result to the DbContext change tracker.
+        //If the result was retrieved from the cache, or it is going to be used by another entity that _was_ retrieved from the cache,
+        //Then we need to ensure that EF Core is aware of it, otherwise it will not fix-up the navigations as it would normally
+        if (result != null)
+        {
+            AttachEntity(result);
+        }
+
+        return result;
+    }
+
+    public void AttachEntity<T>(T entity)
+    {
+        base.Attach(entity);
```

Comment:
I reckon do the is null check in here to stop sonar lint whining and save doing it in both methods above